### ISSUE TITLE
Coach emails not sending

### ISSUE THREAD

**Issue by deniseyu** [codebar/planner#260]
Unclear why coach emails not sending for weekly workshops.

**Comment by despo** [comment]
This is only happening for London. I am guessing that someone who registered as a student doesn't have the email setup - or something similar - and invitation sending falls over. I think some some exception handling to ensure that the method is always called should sort this out.

**Comment by samsworldofno** [comment]
I have a fix for this here: https://github.com/codebar/planner/pull/263

I haven't tested that this is definitely the bug, but I'm pretty sure it is.

**Comment by deniseyu** [comment]
@samsworldofno I'm not sure that fixed it. Invites went out to some coaches today but not all. I think we are getting closer, but I'm still baffled why coach invites are falling over :(

**Comment by samsworldofno** [comment]
Hmmm, ok. I got mine. Do you know who didn't get theirs? I'll try to take a look at the log tomorrow.

**Comment by deniseyu** [comment]
I did not receive one, Spike didn't get one either. It's likely that it errored out quite early in the coaches' list because usually the coach slots would be filled by now!

**Comment by samsworldofno** [comment]
So I can't find anything more wrong with this. Looking at the logs, there are no errors - so it's likely the emails didn't get queued, rather than them failing. I've added a test for the fix earlier in the week, and a small amount of additional logging.

One sensible change would be to remove the batching of this, and queue an individual job for each email. That way if one failed, it would bring down others. Additionally, it would mean we could count the number of jobs created in the test and see if it tallies with the number of emails that we want to send.

If batching was removed, we could also change the `#save` to a `#save!` in `invitation_manager.rb:16`. This would bubble up an error if the coach's invitation couldn't be saved - right now, nothing will happen. Better to have a crash we can see than a silent error we cannot, especially if it will only effect that coach's invitation and not the whole group.
